app/src/main/java/ru/flicksbox/movie/data/MovieRepository.kt

```diff
@@ -0,0 +1,59 @@
+package ru.flicksbox.movie.data
+
+import kotlinx.coroutines.flow.*
+import ru.flicksbox.data.Data
+import ru.flicksbox.movie.domain.*
+import ru.flicksbox.network.BaseDTO
+import ru.flicksbox.network.ResponseErrorManager
+
+class MovieRepositoryImpl(
+    private val movieService: MovieService,
+    private val responseErrorManager: ResponseErrorManager,
+) : MovieRepository {
+    override fun getTopMovies(count: Int, from: Int): Flow<Data<List<MovieEntity>>> =
+        flow { emit(movieService.getTopMovies(count, from)) }
+            .map { response ->
+                responseErrorManager.checkResponseError<BaseDTO<MoviesDto>>(response)
+            }
+            .map { response ->
+                responseErrorManager.checkBodyContainsError<MoviesDto>(response)
+            }
+            .map { movies -> movies.movies.map { it.toDomain() } }
+            .map { Data.content(it) }
+            .catch { emit(Data.error(it)) }
+}
+
+fun MovieDto.toDomain(): MovieEntity =
+    MovieEntity(
+        this.actors?.map { it.toDomain() },
+        this.contentID,
+        this.countries?.map { it.toDomain() },
+        this.description,
+        this.directors?.map { it.toDomain() },
+        this.genres?.map { it.toDomain() },
+        this.contentID,
+        this.images,
+        this.isFavorite,
+        this.isLike,
+        this.isFree,
+        this.name,
+        this.originalName,
+        this.rating,
+        this.shortDescription,
+        this.type,
+        this.video,
+        this.year
+    )
+
+fun MovieDtoActor.toDomain(): MovieEntityActor =
```

Comment:
I suggest to keep `Dto` as postfix, it's much more readable
`MovieActorDto`
`MovieContryDto`
etc